sdks/java/io/thrift/src/main/java/org/apache/beam/sdk/io/thrift/ThriftCoder.java

```diff
@@ -72,15 +71,13 @@ protected ThriftCoder(Class<T> type, TProtocolFactory protocolFactory) {
    */
   @Override
   public void encode(T value, OutputStream outStream) throws CoderException, IOException {
-    ByteArrayOutputStream baos = new ByteArrayOutputStream();
-    TProtocol protocol = protocolFactory.getProtocol(new TIOStreamTransport(baos));
+    TProtocol protocol = protocolFactory.getProtocol(new TIOStreamTransport(outStream));
```

Comment:
@chrlarsen Just trying to make sure I haven't missed anything: was there a reason why we're currently writing to a temporary in-memory stream and then copy that to the actual stream? i.e. do you see any potential pitfalls with the optimization that I just made?

Reply:
LGTM. I think this was an artifact from earlier development.